packages/mattermost-redux/src/selectors/entities/roles.ts

```diff
@@ -50,6 +50,43 @@ export const getMyGroupRoles: (state: GlobalState) => Record<string, Set<string>
     },
 );
 
+export const getGroupListPermissions: (state: GlobalState) => Record<string, GroupPermissions> = createSelector(
+    'getGroupListPermissions',
+    getMyGroupRoles,
+    getRoles,
+    getMySystemPermissions,
+    (state) => state.entities.groups.groups,
+    (myGroupRoles, roles, systemPermissions, allGroups) => {
+        const groups = Object.entries(allGroups).filter((entry) => (entry[1].allow_reference && entry[1].delete_at === 0)).map((entry) => entry[1]);
+
+        const permissions = new Set<string>();
+        groups.forEach((group) => {
+            if (myGroupRoles[group.id!]) {
```

Comment:
a variable (like `roleNames`) could be instantiated with the value of `myGroupRoles[group.id!]` to avoid re-accessing the object.